runtime/cudaq/builder/kernel_builder.h

```diff
@@ -430,6 +451,15 @@ class kernel_builder : public details::kernel_builder_base {
     return details::qalloc(*opBuilder.get(), size);
   }
 
+  // @brief Return a `QuakeValue` representing the allocated
+  // quantum register, initialized to the given state vector.
+  // Note - input argument os not const here, user has to own the data.
+  QuakeValue qalloc(std::vector<cudaq::complex> &state) {
```

Comment:
std::span ?

Reply:
I wonder about that - since we also need to support C++17 now.

Reply:
Shouldn't this actually be `const` so that the compiler doesn't make a temporary?